dbt/adapters/bigquery/clients.py

```diff
@@ -0,0 +1,67 @@
+from google.api_core.client_info import ClientInfo
+from google.api_core.client_options import ClientOptions
+from google.api_core.retry import Retry
+from google.auth.exceptions import DefaultCredentialsError
+from google.cloud.bigquery import Client as BigQueryClient
+from google.cloud.dataproc_v1 import BatchControllerClient, JobControllerClient
+from google.cloud.storage import Client as StorageClient
+
+from dbt.adapters.events.logging import AdapterLogger
+
+import dbt.adapters.bigquery.__version__ as dbt_version
+from dbt.adapters.bigquery.credentials import (
+    BigQueryCredentials,
+    google_credentials,
+    setup_default_credentials,
+)
+
+
+_logger = AdapterLogger("BigQuery")
+
+
+def bigquery_client(credentials: BigQueryCredentials) -> BigQueryClient:
+    try:
+        return _bigquery_client(credentials)
+    except DefaultCredentialsError:
+        _logger.info("Please log into GCP to continue")
+        setup_default_credentials()
+        return _bigquery_client(credentials)
+
+
+@Retry()  # google decorator. retries on transient errors with exponential backoff
+def storage_client(credentials: BigQueryCredentials) -> StorageClient:
```

Comment:
```suggestion
def gcs_storage_client(credentials: BigQueryCredentials) -> StorageClient:
```

Reply:
Same here, are there going to be other storage clients? If so, then sure.

Reply:
it's certainly possible but I'd argue that it reduces mental overhead as you could confuse this with some sort of local file storage client

Reply:
Fair, so then should we go with `gcs_client` to avoid the "PIN number" scenario?